ios/CodePush/CodePushDownloadHandler.m

```diff
@@ -98,13 +112,24 @@ - (void)connection:(NSURLConnection*)connection didFailWithError:(NSError*)error
 }
 
 - (void)connectionDidFinishLoading:(NSURLConnection *)connection {
+    [self.outputFileStream close];
+    
+    // MUST have received bytes from remote server. do not do this, we will get a 0 bytes file of download.zip in download directory,
+    // the variable isZip is FALSE and download.zip will be renamed app.bundle that occurring by self.doneCallback(isZip) method.
+    if (self.receivedContentLength < 1) {
+        NSError *err = [NSError errorWithDomain:@"received nothing"
+                                           code:-1
+                                       userInfo:nil];
+        self.failCallback(err);
```

Comment:
Actually, when got a 400 code, the code
```objective-c
+        if (statusCode >= 400) {
 +            [self.outputFileStream close];
 +            [connection cancel];
 +            
```
 connection will be canceled and didReceiveResponse does not be called.